helix-term/src/ui/editor.rs

```diff
@@ -706,6 +769,315 @@ impl EditorView {
         );
     }
 
+    /// Render the sticky context
+    #[allow(clippy::too_many_arguments)]
+    pub fn render_sticky_context(
+        doc: &Document,
+        view: &View,
+        surface: &mut Surface,
+        context: &Option<Vec<StickyNode>>,
+        doc_annotations: &TextAnnotations,
+        line_decoration: &mut [Box<dyn LineDecoration + '_>],
+        translated_positions: &mut [TranslatedPosition],
+        theme: &Theme,
+    ) {
+        if let Some(context) = context {
+            let text = doc.text().slice(..);
+            let viewport = view.inner_area(doc);
+
+            // backup (status line) shall always exist
+            let status_line_style = theme
+                .try_get("ui.statusline")
+                .expect("`ui.statusline` exists");
+
+            // define sticky context styles
+            let context_style = theme
+                .try_get("ui.sticky.context")
+                .unwrap_or(status_line_style);
+            let indicator_style = theme
+                .try_get("ui.sticky.indicator")
+                .unwrap_or(status_line_style);
+
+            let mut context_area = viewport;
+            context_area.height = 1;
+
+            for node in context {
+                surface.clear_with(context_area, context_style);
+
+                let mut new_offset = view.offset;
+                let mut line_context_area = context_area;
+
+                if let Some(indicator) = node.indicator.as_deref() {
+                    // set the indicator
+                    surface.set_stringn(
+                        line_context_area.x,
+                        line_context_area.y,
+                        indicator,
+                        indicator.len(),
+                        indicator_style,
+                    );
+                    continue;
+                }
+
+                // get the len of bytes of the text that will be written (the "definition" line)
+                let line = text.line(node.line);
+                let tab_width_count = line.chars().filter(|c| *c == '\t').count();
+
+                let already_written =
+                    (line.len_bytes() + tab_width_count.saturating_mul(doc.tab_width() - 1)) as u16;
```

Comment:
Is using `line.len_bytes()` here correct? Won't this shows up incorrectly for CJK characters?

Reply:
I figured, that it may be the best idea to use bytes, as:
- I am already calculating with byte sizes
- Fullwidth characters would disallow me from using character lens, as they are definetly larger than a single unit.

Have you another (better) suggestion?